src/components/PopupWithForm.js

```diff
@@ -24,6 +28,14 @@ export default class PopupWithForm extends Popup {
     this._popupForm.reset();
   }
 
+  setLoading(submit, loadingText = "Saving...") {
+    if (submit) {
+      this._submitButton.textContent = loadingText;
+    } else {
+      this._submitButton.textContent = this._submitButtonText;
+    }
+  }
+
```

Comment:
**Could be improved**

You can make ` setInputValues` in class `PopupWithForm`, which can insert data into inputs:

```js

  setInputValues(data) {
    this._inputList.forEach((input) => {
      // here you insert the `value` by the `name` of the input
      input.value = data[input.name];
    });
  }



```

So, this way you’ll not have to search the inputs of the profile in `index.js`